### PR TITLE
Prometheus

### PR DESCRIPTION
**Pull request by max-jacobs**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -206,3 +206,9 @@ variable "docker_registry_auth_secret_name" {
   type        = "string"
   default     = ""
 }
+
+variable "prometheus_enabled" {
+  description = "Enable to expose the Prometheus endpoint. Also enables Prometheus metrics scraping from k8s pods - true or false."
+  default     = "false"
```
Comment (rpoluri) [inline variables.tf]: can you use `type = bool` and set `default = false` without quotes, new terraform versions shows warning with old syntax.

File: templates.tf [hunk 2]
```diff
@@ -19,7 +32,8 @@ data "template_file" "server_yaml" {
   template = file("${path.module}/templates/waggle-dance-server.yml.tmpl")
 
   vars = {
-    graphite = join("", data.template_file.graphite_server_yaml.*.rendered)
+    graphite          = join("", data.template_file.graphite_server_yaml.*.rendered)
+    exposed_endpoints = join("", data.template_file.endpoints_server_yaml.rendered)
```
Comment (rpoluri) [inline templates.tf]: may not need join here, join was mainly use to make blocks optional in terraform 11, seems like we will be configuring default enpoint list when prometheus is not enabled.
exposed_endpoints = join("", data.template_file.endpoints_server_yaml.rendered)

so we can just setup as
exposed_endpoints = data.template_file.endpoints_server_yaml.rendered

also FYI we can refactor waggle-dance server yaml using if conditions in a single file in terraform 12, I will create another PR for that.

File: variables.tf (hunk 1 above)
Comment (max-jacobs) [reply variables.tf]: Thanks, will do

File: templates.tf (hunk 2 above)
Comment (max-jacobs) [reply templates.tf]: Make sense. Do you have any idea if there are any other endpoints in use, aside from the one in the default list here?

**Comment by max-jacobs** [comment]
@patduin @rpoluri I can confirm that this is now working, please can I get a re-approval to merge? thanks!

**Pull request merged**